app/org/zalando/markscheider/MetricsController.scala

```diff
@@ -2,26 +2,70 @@ package org.zalando.markscheider
 
 import java.io.StringWriter
 
-import play.api.mvc.{ Action, AnyContent, InjectedController, Result }
+import com.codahale.metrics.MetricRegistry
 import com.fasterxml.jackson.databind.{ ObjectMapper, ObjectWriter }
 import com.google.inject.Inject
+import play.api.Configuration
 import play.api.http.{ HeaderNames, MimeTypes }
+import play.api.mvc._
 
-class MetricsController @Inject() (plugin: MetricsPlugin) extends InjectedController {
+import scala.collection.JavaConverters._
 
-  def serialize(mapper: ObjectMapper): Result = {
+class MetricsController @Inject() (
+    plugin:        MetricsPlugin,
+    configuration: Configuration,
+    cc:            ControllerComponents
+)
+  extends AbstractController(cc) with PrometheusSupport {
+  private def defaultFormat: String = configuration.getAndValidate[String]("org.zalando.markscheider.defaultFormat", Set("zmon", "prometheus"))
+
+  def serializePrometheus(registry: MetricRegistry): Result = {
+    val histograms = (for {
+      (metricsName, histogram) <- registry.getHistograms.asScala
+      extractedName <- ExtractedName.fromMetricsName(metricsName)
+    } yield {
+      import extractedName.{ labels, name }
+
+      PrometheusMetric.fromSnapshot(name, labels, histogram.getSnapshot)
+    }).flatten
+
+    val timers = (for {
+      (metricsName, timer) <- registry.getTimers().asScala
+      extractedName <- ExtractedName.fromMetricsName(metricsName)
+    } yield {
+      import extractedName.{ labels, name }
+
+      PrometheusMetric.fromTimer(name, labels, timer)
+    }).flatten
+
+    val stringWriter = new StringWriter()
+    for { entry <- histograms ++ timers } {
+      stringWriter.write(entry.toString)
+    }
+    Ok(stringWriter.toString).withHeaders(HeaderNames.CACHE_CONTROL -> "must-revalidate,no-cache,no-store").as("text/plain; version=0.0.4")
+  }
+
+  def serializeJson(mapper: ObjectMapper): Result = {
```

Comment:
This one could be private, right?

```suggestion
  private def serializeJson(mapper: ObjectMapper): Result = {
```

Reply:
yep